### PR TITLE
tests for `/token/refresh` and POST `/token` with `grant_type=refresh_token`

### LINKED ISSUE #274: Tests for token refresh.
- Re-use/adopt services used in /token/refresh endpoint (introduced in https://github.com/fabric8-services/fabric8-auth/pull/265) in /token endpoint
- Cover both endpoints by automated tests (do not test Keycloak! Test our controllers only. Mock the service which is supposed to call KC)

### PR DESCRIPTION
**Pull request by kishansagathiya**
Fixes #274

### CODE REVIEW

**Comment by fabric8cd** [comment]
@kishansagathiya snapshot fabric8-auth image is available for testing.  `docker pull docker.io/fabric8/fabric8-auth:SNAPSHOT-PR-310-1`

File: controller/token_blackbox_test.go [hunk 1]
```diff
@@ -98,6 +99,35 @@ func (rest *TestTokenREST) TestRefreshTokenUsingNilTokenFails() {
...
+		RefreshToken: &refreshToken,
+	}
+	_, authToken := test.RefreshTokenOK(t, service.Context, service, controller, payload)
+	token := authToken.Token
+	require.NotNil(rest.T(), token.TokenType)
+	require.Equal(rest.T(), "bearer", *token.TokenType)
+	require.NotNil(rest.T(), token.AccessToken)
+	require.Equal(rest.T(), rest.sampleAccessToken, *token.AccessToken)
+	require.NotNil(rest.T(), token.RefreshToken)
+	require.Equal(rest.T(), rest.sampleRefreshToken, *token.RefreshToken)
```
Comment (alexeykazakov) [inline controller/token_blackbox_test.go]: Check the expire_in too.
And move all these checks to a new function and reuse it in both TestRefreshTokenUsingCorrectRefreshTokenOK() and checkExchangeWithRefreshToken()
```
	require.NotNil(rest.T(), token.TokenType)
 	require.Equal(rest.T(), "bearer", *token.TokenType)
 	require.NotNil(rest.T(), token.AccessToken)
 	require.Equal(rest.T(), rest.sampleAccessToken, *token.AccessToken)
 	require.NotNil(rest.T(), token.RefreshToken)
 	require.Equal(rest.T(), rest.sampleRefreshToken, *token.RefreshToken)
 	expiresIn, err := strconv.Atoi(*token.ExpiresIn)
 	require.Nil(rest.T(), err)
 	require.True(rest.T(), expiresIn > 60*59*24*30 && expiresIn < 60*61*24*30) // The expires_in should be withing a minute range of 30 days.
```

**Comment by fabric8cd** [comment]
@kishansagathiya snapshot fabric8-auth image is available for testing.  `docker pull docker.io/fabric8/fabric8-auth:SNAPSHOT-PR-310-2`

**Comment by fabric8cd** [comment]
@kishansagathiya snapshot fabric8-auth image is available for testing.  `docker pull docker.io/fabric8/fabric8-auth:SNAPSHOT-PR-310-3`

File: controller/token_blackbox_test.go (hunk 1 above)
Comment (alexeykazakov) [reply controller/token_blackbox_test.go]: OK, I see that these are two different types, so, it's tricky to re-use the same function. And I've added the check for expireIn.

**Pull request merged**